types/uint.go

```diff
@@ -0,0 +1,173 @@
+package types
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/pkg/errors"
+)
+
+// Unt wraps integer with 256 bit range bound
+// Checks overflow, underflow and division by zero
+// Exists in range from 0 to 2^256-1
+type Uint struct {
+	i *big.Int
+}
+
+// NewUintFromBigUint constructs Uint from big.Uint
+func NewUintFromBigInt(i *big.Int) Uint {
+	u, err := checkNewUint(i)
+	if err != nil {
+		panic(fmt.Errorf("overflow: %s", err))
+	}
+	return u
+}
+
+// NewUint constructs Uint from int64
+func NewUint(n uint64) Uint {
+	i := new(big.Int)
+	i.SetUint64(n)
+	return NewUintFromBigInt(i)
+}
+
+// NewUintFromString constructs Uint from string
+func NewUintFromString(s string) Uint {
+	u, err := ParseUint(s)
+	if err != nil {
+		panic(err)
+	}
+	return u
+}
+
+// ZeroUint returns unsigned zero.
+func ZeroUint() Uint { return NewUint(0) }
```

Comment:
Do we need to run through all the checks for 0 and 1?
Maybe it would be more efficient to avoid them... specially as this will affect `IsZero`

Reply:
Fixed